### ISSUE TITLE
Parameters initialization for RelGAN_D?

### ISSUE THREAD

**Issue by lethaiq** [williamSYSU/TextGAN-PyTorch#21]
From the original code of RelGAN (https://github.com/weilinie/RelGAN), D is initialized with truncated_normal (similar with G). In your code, you are using Uniform (I saw a commit that you did so). Is it a preference or I am missing something? I am trying to figure out where I got wrong in the NLL_gen curve (https://github.com/williamSYSU/TextGAN-PyTorch/issues/20), so I want to ask about this.

Thanks!

**Comment by williamSYSU** [comment]
Indeed, the discriminator D is initialized with `uniform` in my implemented codes. The reason is that the NLL_oracle only decreases under the `uniform` initialization, while continuously increases under the `truncated_normal` initialization. I don't know why, it may be caused by the differences between PyTorch and Tensorflow.

**Comment by lethaiq** [comment]
> Indeed, the discriminator D is initialized with `uniform` in my implemented codes. The reason is that the NLL_oracle only decreases under the `uniform` initialization, while continuously increases under the `truncated_normal` initialization. I don't know why, it may be caused by the differences between PyTorch and Tensorflow.

Uhm interesting cause in original RelGAN code it only uses truncated_normal. Have you tried with a different random_seed?

**Comment by williamSYSU** [comment]
Besides, I also try uniform initialization in original RelGAN code, all metrics are normal and are not influenced by the ways of initialization. 

> Have you tried with a different random_seed?

Actually, there is no seed in my codes. So it means each run of program will initialize different training weights.

**Comment by lethaiq** [comment]
> Besides, I also try uniform initialization in original RelGAN code, all metrics are normal and are not influenced by the ways of initialization.
> 
> > Have you tried with a different random_seed?
> 
> Actually, there is no seed in my codes. So it means each run of program will initialize different training weights.

Thanks. Then the initialization is not the main cause....

**Comment by williamSYSU** [comment]
Yes, although the initialization affects performances, it is not the cause of the divergence of NLL_gen.
